noir-projects/aztec-nr/aztec/src/context/public_context.nr

```diff
@@ -130,9 +131,16 @@ impl PublicContext {
         self.return_hash = returns_hasher.hash();
     }
 
+
+    fn push_unencrypted_log(&mut self, log_hash: Field) {
```

Comment:
There's already an `accumulate_unencrypted_logs`. Is this different or should that one be removed?

In any case, please make the changes to PublicContextInterface, and please make sure that the AvmContext also implements it correctly.

PublicContext (and all the current public implementation) will be replaced by the AvmContext (and the AVM). I created the PublicContextInterface to keep them in sync.

Reply:
> There's already an accumulate_unencrypted_logs. Is this different or should that one be removed?

I wasn't sure whether we wanted to use it once we hash the raw logs inside the circuits (1165), and the `push` is just an interim. I see arguments either way.

I can add `push_unencrypted_logs` to `PublicContextInterface` but I'm a little lost on where it should go - `ContextInterface` has `push_new_note_hash` which also pushes a new side effect, but `PublicContextInterface` has accumulate logs - including encrypted logs, which are never used for `PublicContext`. Any pointers?

Reply:
If this is something common to all contexts (private, public, avm) then it can go in ContextInterface. Otherwise, if it's just common to public ones, it can go in PublicContextInterface.

I'll do some clean up later so don't worry too much :)

Reply:
So I have added it to public - it's a strange one, because the avm context actually works as intended (I think). As in it goes:
Contract emits log -> context ->`emit_unencrypted_log` -> oracle -> state updated in avm
However, private and public go:
Contract emits log -> oracle ->`emit_unencrypted_log` -> returns log hash -> context -> state updated via `push_(un)encrypted_logs`
So I think once 1165 is done, we just want `emit` and no `push`, since `emit` should do both jobs (hash and update state). This is why the update is a little janky. I'm happy to change it up since it's temporary anyway!

Reply:
Should be all good now - the e2e bench failures are from master and I don't have anything outstanding from this task :)